### ISSUE TITLE
Bugfix: json_dump can not encode numpy data types

### ISSUE THREAD

**Issue by pziarsolo** [zarr-developers/zarr-python#569]
#### Minimal, reproducible code sample, a copy-pastable example if possible

```python

import numpy
import dask.array
import zarr


def main():
    np_array = numpy.random.randint(1, 10, size=1000000)
    array = dask.array.from_array(np_array)

    array = array[array > 5]

    with TemporaryDirectory() as tmpdir:
        array.compute_chunk_sizes()
        delayed = dask.array.to_zarr(array, url=tmpdir, compute=False,
                                                      component='/data')
        dask.compute(delayed)

        z_object = zarr.open_group(tmpdir, mode='r')

        assert numpy.all(np_array[np_array > 5] == z_object.data[:])


if __name__ == '__main__':
    main()


```

#### Problem description
dask array.compute_chunk_size() generates chunks with numpy.int64 integers.

When zarr tries to encode dask array metadata into json format, it  fails because default json encoder does not understand numpy data types.

Giving an numpy  data type encoder to json.dump solves the problem

#### Version and installation information

Please provide the following:
zarr.__version__ 2.4
numcodecs      0.6.4
python 3.8
Operating system: Linux
How Zarr was installed: pip into virtual environment

**Comment by pziarsolo** [comment]
pull request #570 solves the problem

**Comment by jakirkham** [comment]
> dask array.compute_chunk_size() generates chunks with numpy.int64 integers.

In that case, I think the right thing to do would be to fix this in Dask. I'm pretty sure this is not intentional behavior. Could you please raise an issue on [the Dask issue tracker]( https://github.com/dask/dask/issues ) or better yet add a PR over there? Thanks for letting us know about this issue 🙂

**Comment by pziarsolo** [comment]
> > dask array.compute_chunk_size() generates chunks with numpy.int64 integers.
> 
> In that case, I think the right thing to do would be to fix this in Dask. I'm pretty sure this is not intentional behavior. Could you please raise an issue on [the Dask issue tracker](https://github.com/dask/dask/issues) or better yet add a PR over there? Thanks for letting us know about this issue slightly_smiling_face

Ok.
I will create and issue in dask
thanks

**Comment by alimanfoo** [comment]
Just to second @jakirkham, probably better to get dask to convert this back into plain Python ints.

**Comment by jakirkham** [comment]
This was fixed in Dask 2.19.0 with PR ( https://github.com/dask/dask/pull/6326 ). Going to go ahead and close. If there are other problems, please feel free to raise a new issue. Thanks!
